core/src/components/loading/loading.tsx

```diff
@@ -158,11 +253,22 @@ export class Loading implements ComponentInterface, OverlayInterface {
    * Some examples include: ``"cancel"`, `"destructive"`, "selected"`, and `"backdrop"`.
    */
   @Method()
-  dismiss(data?: any, role?: string): Promise<boolean> {
+  async dismiss(data?: any, role?: string): Promise<boolean> {
     if (this.durationTimeout) {
       clearTimeout(this.durationTimeout);
     }
-    return dismiss(this, data, role, 'loadingLeave', iosLeaveAnimation, mdLeaveAnimation);
+    this.currentTransition = dismiss(this, data, role, 'loadingLeave', iosLeaveAnimation, mdLeaveAnimation);
+
+    const dismissed = await this.currentTransition;
+
+    if (dismissed) {
+      const { delegate } = this.delegateController.getDelegate();
+      if (delegate) {
+        await detachComponent(delegate, this.usersElement);
+      }
+    }
+
+    return dismissed;
```

Comment:
I do not understand the purpose of this code. `this.usersElement` should only be defined in components that accept user-defined children such as in `ion-modal` and `ion-popover`. `ion-loading` does not accept user-defined children, so this should not be needed.

Reply:
I can refactor this. The inline overlays require that `attachViewToDom`/`removeViewFromDom` is called when presenting/dismissing; otherwise the overlay cannot be re-presented after being dismissed.

The `detachComponent` utility handled this behavior, but also deals with the user's provided component, which isn't needed here.